README.md

```diff
@@ -2,73 +2,250 @@
+
+When calling the `docsSearchBar` method, you can add optional fields:
+
+##### `queryHook` <!-- omit in toc -->
+
+`queryHook` takes a callback function as value. This function will be called on every keystroke to transform the typed keywords before querying MeiliSearch. By default, it does not do anything.
+
+##### `transformData` <!-- omit in toc -->
+
+`transformData` takes a callback function as value. This function will be called on every hit before displaying them. By default, it does not do anything.
+
+##### `handleSelected` <!-- omit in toc -->
+
+`handleSelected` takes a callback function a value. This function is called when a suggestion is selected (either from a click or a keystroke). By default, it displays anchor links to the results page. Here is an example to override this behavior:
+
+```javascript
+docsSearchBar({
+  // ...
+  handleSelected: function(input, event, suggestion, datasetNumber, context) {
+    // Prevents the default behavior on click and rather opens the suggestion
+    // in a new tab.
+    if (context.selectionMethod === "click") {
+      input.setVal("");
+
+      const windowReference = window.open(suggestion.url, "_blank");
+      windowReference.focus();
+    }
+  }
+});
+```
+
+Note that, by default, you can already open a new tab thanks to the CMD/CTRL + Click action.
+
+The function is called with the following arguments:
+
+- `input`: a reference to the search input element. It comes with the `.open()`, `.close()`, `.getVal()` and `.setVal()` methods.
+
+- `event`: the actual event triggering the selection.
+
+- `suggestion`: the object representing the current selection. It contains a `.url` key representing the destination.
+
+- `datasetNumber`: this should always be equal to 1 as `docs-searchbar.js` is searching into one dataset at a time. You can ignore this attribute.
+
+- `context`: additional information about the selection. Contains a `.selectionMethod` key that can be either `click`, `enterKey`, `tabKey` or `blur`, depending on how the suggestion was selected.
+
+##### `meilisearchOptions` <!-- omit in toc -->
+
+You can forward search parameters to the MeiliSearch API by using the `meilisearchOptions` key. Checkout out the [MeiliSearch documentation about search parameters](https://docs.meilisearch.com/guides/advanced_guides/search_parameters.html#search-parameters).
+
+For example, you might want to increase the number of results displayed in the dropdown:
+
+```javascript
+docsSearchBar({
+  meilsearchOptions: {
+    limit: 10
+  }
+});
+```
+
+##### More Examples <!-- omit in toc -->
+
+Here is a basic [HTML file](scripts/playground.html) used in the playground of this repository.
+
+As a more concrete example, you can [check out the configuration](https://github.com/meilisearch/vuepress-plugin-meilisearch/blob/bf2210e131eb5630f632d514f35710ee6576ec7a/MeiliSearchBox.vue#L64) applied in the MeiliSearch plugin for Vuepress.
+
+#### Styling <!-- omit in toc -->
```

Comment:
```suggestion
##### Styling <!-- omit in toc -->
```

Reply:
There is a misunderstanding. The `More examples` section is a sub-title of `Optional Parameters` section. But `Styling` is not, it's the same level than `Optional Parameters`.